apps/framework-cli/src/infrastructure/redis/redis_client.rs

```diff
@@ -164,32 +164,23 @@ impl RedisClient {
     }
 
     pub async fn register_lock(&mut self, name: &str, ttl: i64) -> Result<()> {
+        info!("<RedisClient> Registering lock {}", name);
         let lock_key = self.service_prefix(&["lock"]);
         let lock = RedisLock { key: lock_key, ttl };
         self.locks.insert(name.to_string(), lock);
         Ok(())
     }
 
     pub async fn attempt_lock(&self, name: &str) -> Result<bool> {
+        info!("<RedisClient> Attempting to lock {}", name);
         if let Some(lock) = self.locks.get(name) {
-            let result: bool = self
-                .connection
-                .lock()
-                .await
+            let result = redis::pipe()
+                .atomic()
                 .set_nx(&lock.key, &self.instance_id)
```

Comment:
what about using `SetOptions` with `ExistenceCheck::NX` and `SetExpiry::EX(60)`?

Reply:
didn't try or test that approach. We can certainly do that in a new test PR